### DISCUSSION TITLE
How do we update the content item programmatically.

### DISCUSSION THREAD

**Discussion by mrpavlitzky** [OrchardCMS/OrchardCore#12173] [Q&A] [answered]
Hello,

I've seen many resources online on how to create content items programatically, but I couldn't find any info on updating existing items' fields. I'd be grateful for any tips.

Cheers

**Comment by lampersky** [comment, marked as answer]
@Cinnam00n 
let's say you have **MyPart** added to the content item:

```
        public class MyPart : ContentPart
        {
            public MyPart() { }
            public TextField Name { get; set; }
            public MediaField Image { get; set; }
        }
```

you can modify content item this way:

```
        public async Task UpdateContentItem(string contentItemId, string newImagePath, string newName)
        {
            var contentItem = await _contentManager.GetAsync(contentItemId);

            contentItem.Alter<MyPart>(a =>
            {
                a.Image = new MediaField { Paths = new string[] { newImagePath } };
                a.Name = new TextField { Text = newName };
            });

            await _contentManager.UpdateAsync(contentItem);
        }
```

**Reply by mrpavlitzky** [reply]
@lampersky oh, thank you. I didn't know about Alter method is it documented anywhere?

**Reply by lampersky** [reply]
I'm not familiar with the whole documentation, but there is a chance, this is not documented yet.

maybe it would be good to extend this section a bit: https://docs.orchardcore.net/en/dev/docs/reference/modules/ContentTypes/

**Comment by hyzx86** [comment]
You can try  workflow 

![image](https://user-images.githubusercontent.com/15613121/184902515-49c04ab1-b1b6-44cb-9c0a-9221891644f2.png)

this is my project ,witch using `Create Content`


![image](https://user-images.githubusercontent.com/15613121/184902249-4f8ef0c9-aa19-4623-be20-2bd5b5865bd5.png)

**Reply by hyzx86** [reply]
this is a sample
![image](https://user-images.githubusercontent.com/15613121/184904313-cc76f884-916c-4269-a6b6-a3fc94c6035f.png)

change the extension to .zip

[test.zip.txt](https://github.com/OrchardCMS/OrchardCore/files/9351896/test.zip.txt)

**Reply by mrpavlitzky** [reply]
Huh, very interesting I'll surely dig into it in the future. I haven't really thought about using workflows, but it looks like a neat solution.

**Comment by jtkech** [comment]
@hyzx86 is the workflows guru ;)

**Reply by hyzx86** [reply]
😁OC's workflow is very powerful. If you can understand it better, you will get twice the result with half the effort

**Comment by jtkech** [comment]
Forgot to mention another great WF guru => @lampersky !!!
